### ISSUE TITLE
Buttons/icons replaced by garbage data

### ISSUE THREAD

**Issue by volumetricsteve** [bbidulock/icewm#102]
the buttons that appear on the upper right of each window appear to be pulling in garbage data from the frame buffer.  I can provide pictures if necessary.

**Comment by bbidulock** [comment]
Yes, please provide pictures.  Also, identify which theme you are using (hopefully one of those installed with icewm).

**Comment by volumetricsteve** [comment]
is there a built in screenshot system in xorg/icewm?  if not, I can just snap some with my phone.  I'm using whatever the bare-bones default is, i'm not customizing anything when I report these issues.

**Comment by volumetricsteve** [comment]
Here's 3 quick instances.

http://imgur.com/a/8PPMq

**Comment by bbidulock** [comment]
You can use xwd.

**Comment by volumetricsteve** [comment]
xwd isn't provided by my distro, so I'll have to do some digging.  are the screenshots I provided at least good enough to get an idea?

**Comment by bbidulock** [comment]
See issue #101 and comments on configure options.  Neither shaped decorations nor gradients will display correctly unless the options are enabled.

**Comment by volumetricsteve** [comment]
I think we discussed this elsewhere, but I wanted to pass on that I have the aforementioned options enabled.  However, I realized the issue I have only happens with a few themes.  The default one being one of them.  I found one that wasn't impacted by this (there are a handful) and that's my workaround...  Based on other research I've done, it seems like some themes use a 1 pixel thin image, which doesn't render on certain nvidia devices in X.  This can't be true for the buttons though, so I still don't know what's going on there.

**Comment by gijsbers** [comment]
Something very similar occurs for me with the IceGraphite theme from:
https://www.box-look.org/p/1119202/
The title bar icons are messed up. Some of the cursors and network and CPU monitoring pixmaps are shown as title bar icons. This theme also has 1 pixel wide title bar XPM icons.

**Comment by gijsbers** [comment]
Fixed by commit 4b54eebccf231b24798ff988052bc9d15f2d47d3

**Comment by bbidulock** [comment]
What part of 4b54eeb fixed this?

**Comment by gijsbers** [comment]
The else without brace needed a brace in:
```
@@ -266,14 +275,19 @@ void YFrameButton::paint(Graphics &g, const YRect &/*r*/) {
                int y(((int)height() - (int)pixmap->height()) / 2);
                g.drawPixmap(pixmap, x, y);
             }
-        } else
+        }
+        else {
             g.fillRect(0, 0, width(), height());
 
-        if (fAction == 0 && icon != null && showFrameIcon)
-            icon->draw(g,
-                       ((int)width() - (int)iconSize) / 2,
-                       ((int)height() - (int)iconSize) / 2,
-                       iconSize);
+#ifndef LITE
+            if (fAction == 0 && icon != null && showFrameIcon) {
+                icon->draw(g,
+                           ((int)width() - (int)iconSize) / 2,
+                           ((int)height() - (int)iconSize) / 2,
+                           iconSize);
+            }
+#endif
+        }
     }
 }
```

**Comment by volumetricsteve** [comment]
For what it's worth, I worked around this by flipping though themes until one of them didn't do it.  I think I read somewhere that single pixel thick bitmaps don't render right because of some ancient X thing being depreciated.  I just needed a theme that didn't have any single-pixel thick images and everything seemed fine.

**Comment by gijsbers** [comment]
Thanks for the feedback @volumetricsteve. In some other issue the single-pixel image problem hypothesis was also raised. When looking for new themes to include in icewm I found some themes with single-pixel images which worked fine. Which made me wonder if that is really the cause. Then later I found other causes which I could fix. What would be helpful is some names of themes which don't work and which do so we can test and examine them.
